README.md

```diff
@@ -2,31 +2,448 @@
+            if (c == 'A') {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    public static void main(String[] args) {
+        System.out.println(countA("BABY")); //1
+        System.out.println(countA("BANANA")); //3
+        System.out.println(countA("123!")); //0
+        System.out.println(countA("")); //0
+        System.out.println(countA("amazing")); //0
+    }
+
+}
+```
+
+The program should print:
+
+```text
+1
+3
+0
+0
+0
+```
+
+Rather than relying on visual inspection of the output produced by the `main` method,
+we should create a Junit class `StringSearchUtilityTest` to test the value
+returned by the `StringSearchUtility.countA` method (we preface the class name since it is
+a static method in another class).
+
+```java
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StringSearchUtilityTest {
+
+    @Test
+    public void countA() {
+        assertEquals(1, StringSearchUtility.countA("BABY"));
+        assertEquals(3, StringSearchUtility.countA("BANANA"));
+        assertEquals(0, StringSearchUtility.countA("123!"));
+        assertEquals(0, StringSearchUtility.countA(""));
+        assertEquals(0, StringSearchUtility.countA("amazing"));
+    }
+
+}
+```
+
+Running the Junit test class `StringSearchUtilityTest` should result in the test passing.
+
+
+## Testing for the presence of a character in a string
+
+Let's add a predicate method `containsA` that returns a `boolean` value indicating
+whether a string contains the character 'A'.  In a subsequent lesson we'll see there
+are existing `String` methods that we could use, but for now we'll implement this
+manually to show an example method that returns a value from within a loop, as well
+as a different value after the loop.
+
+One approach to implement the `containsA` method is to simply have it
+call the `countA` method and test if the result is positive.  However,
+what if some of our strings contain lots of characters?  Rather than iterating through
+the long sequence of characters to count the total number of 'A' characters,
+it would be more efficient to return `true` as soon as we find the first 'A'.
+
+We'll add our new method `containsA` to the `StringSearchUtility` class.
+Note that the method returns `true` in the for loop as soon as it finds an 'A',
+but only returns `false` after looping through the entire
+string and not finding any 'A' characters.
+
+
+```java
+public static boolean containsA(String str) {
+    for (char c : str.toCharArray()) {
+        if (c == 'A') {
+            return true;
+        }
+    }
+    return false;
+}
+```
+
+NOTE: A common error is to use an `if-else` as shown below.  But this code will only
+loop once, returning a value based on whether the
+first character in the string is an 'A', it never looks
+at the rest of the string.  The correct
+code shown above waits until *after* the loop to return `false`.
```

Comment:
This sentence is a little weird. Maybe:

"The correct code shown above waits until _after_ the loop has finished executing to return `false`."

Reply:
fixed